src/actions/api.js

```diff
@@ -1128,25 +1125,61 @@ export const onUserProposalCredits = () => (dispatch, getState) => {
     });
 };
 
-export const onFetchUserProposalsWithVoteSummary = (userid, token) => async (
-  dispatch
-) => {
-  dispatch(act.REQUEST_USER_PROPOSALS({ userid }));
-  try {
-    const { proposals, ...response } = await api.userProposals(userid, token);
-    const publicPropsTokens = proposals
-      .filter((prop) => prop.status === PROPOSAL_STATUS_PUBLIC)
-      .map((prop) => prop.censorshiprecord.token);
-
-    if (publicPropsTokens.length) {
-      await dispatch(onFetchProposalsBatchVoteSummary(publicPropsTokens));
+export const onFetchUserProposals = (userid) =>
+  withCsrf(async (dispatch, getState, csrf) => {
+    dispatch(act.REQUEST_USER_PROPOSALS({ userid }));
+    console.log("onFetchUserProposals");
+    try {
+      const response = await api.userProposals(csrf, userid);
+      console.log("response");
+      console.log(response);
+      const cachedUserProposals = sel.makeGetUserProposals(userid)(getState());
+      const unvettedTokens = Object.values(response.unvetted)
+        .flat(1)
+        .map((t) => ({ token: t }));
+      const vettedTokens = Object.values(response.vetted)
+        .flat(1)
+        .map((t) => ({ token: t }));
+      const tokensLength = unvettedTokens.length + vettedTokens.length;
+
+      if (cachedUserProposals.length === tokensLength) {
+        // props are already loaded on cache
+        return dispatch(act.RECEIVE_USER_PROPOSALS());
+      }
+
+      let unvettedProposals, vettedProposals;
```

Comment:
```suggestion
      let unvettedProposals = [], vettedProposals = [];
```
This fixes an error thrown by `onFetchUserProposals` when there are no unvetted tokens. If no tokens are available, accessing the `unvettedProposals[0]` will lead to an error. The same will happen for vetted proposals.

Reply:
Sweet nice find mate